### PR TITLE
feat: Add cloudwatch lambda event support to Parser utility

### PR DESCRIPTION
**Pull request by risenberg-cyberark**
see #230

### CODE REVIEW

File: aws_lambda_powertools/utilities/parser/models/cloudwatch.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class CloudWatchLogsData(BaseModel):
+    decoded_data: CloudWatchLogsDecode = Field(None, alias="data")
+
+    @validator("decoded_data", pre=True)
+    def prepare_data(cls, value):
+        try:
+            payload = base64.b64decode(value)
+            uncompressed = zlib.decompress(payload, zlib.MAX_WBITS | 32)
+            return json.loads(uncompressed.decode("UTF-8"))
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/parser/models/cloudwatch.py:33]:
```suggestion
            return json.loads(uncompressed.decode("utf-8"))
```

File: tests/functional/parser/test_cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import pytest
+
+from aws_lambda_powertools.utilities.parser import ValidationError, envelopes, event_parser
+from aws_lambda_powertools.utilities.parser.models import CloudWatchLogsLogEvent, CloudWatchLogsModel
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from tests.functional.parser.schemas import MyCloudWatchBusiness
+from tests.functional.parser.utils import load_event
+
+
+@event_parser(model=MyCloudWatchBusiness, envelope=envelopes.CloudatchEnvelope)
```
Comment (heitorlessa) [inline tests/functional/parser/test_cloudwatch.py:15]:
```suggestion
@event_parser(model=MyCloudWatchBusiness, envelope=envelopes.CloudatchLogsEnvelope)
```

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Looks great! Minor suggestions and one question on zlib header/trailer size param.

As with other PRs, I'll write the docs afterwards ;)

File: aws_lambda_powertools/utilities/parser/models/cloudwatch.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class CloudWatchLogsData(BaseModel):
+    decoded_data: CloudWatchLogsDecode = Field(None, alias="data")
+
+    @validator("decoded_data", pre=True)
+    def prepare_data(cls, value):
+        try:
+            payload = base64.b64decode(value)
+            uncompressed = zlib.decompress(payload, zlib.MAX_WBITS | 32)
```
[the remark this answers is not in the document]
Comment (risenberg-cyberark) [reply aws_lambda_powertools/utilities/parser/models/cloudwatch.py:32]: didnt work without it

**Review by heitorlessa — APPROVED** [review]
Docs added and envelope type fixed - Merging

**Pull request merged**